### PR TITLE
Change FUSE readwriteLock to session based lock

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Fix the bug introduced by https://github.com/Alluxio/alluxio/pull/16377.
Change thread based lock to session based lock to avoid IllegalMonitorException when a thread tries to unlock a lock locked by another thread.

### Why are the changes needed?

Fuse stream operations can be called by different threads. stream creation (lock the path) can be called by one thread while stream close (unlock the path) may be called by another thread. Use the previous ReadWriteLock, unlock will throw IllegalMonitorException.

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

File: core/common/src/main/java/alluxio/concurrent/ClientRWLock.java [hunk 1]
```diff
@@ -29,22 +26,25 @@
  */
 @ThreadSafe
 public final class ClientRWLock implements ReadWriteLock {
-  /** Total number of permits. This value decides the max number of concurrent readers. */
-  private static final int MAX_AVAILABLE =
-          Configuration.getInt(PropertyKey.WORKER_TIERED_STORE_BLOCK_LOCK_READERS);
   /**
    * Uses the unfair lock to prevent a read lock that fails to release from locking the block
    * forever and thus blocking all the subsequent write access.
    * See https://alluxio.atlassian.net/browse/ALLUXIO-2636.
```
Comment (beinan) [inline core/common/src/main/java/alluxio/concurrent/ClientRWLock.java:32]: nit: I don't thinks this link is helpful,  remove it?

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java [hunk 2]
```diff
@@ -106,13 +104,7 @@ public final class AlluxioJniFuseFileSystem extends AbstractFuseFileSystem
...
   private static final IndexDefinition<FuseFileEntry<FuseFileStream>, String>
       PATH_INDEX = IndexDefinition.ofUnique(FuseFileEntry::getPath);
-  /**
-   * Making sure only one write stream at a time.
-   * Key is the hash string of file path.
-   */
-  private static final LockPool<String> PATH_LOCKS
-      = new LockPool<>((key) -> new ReentrantReadWriteLock(),
-      128, 128, 512, 64);
+
```
Comment (beinan) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:107]: nit: probably this new line is not necesary

**Review by beinan — APPROVED** [review]
lgtm

**Comment by LuQQiu** [comment]
Error: 7.879 [ERROR] alluxio.server.ft.journal.raft.EmbeddedJournalIntegrationTestTransferLeadership.resetPriorities
Error: 7.880 [ERROR]   Run 1: EmbeddedJournalIntegrationTestTransferLeadership.resetPriorities:207->transferAndWait:250->EmbeddedJournalIntegrationTestBase.waitForQuorumPropertySize:58 » Timeout
Error: 7.884 [ERROR]   Run 2: EmbeddedJournalIntegrationTestTransferLeadership.resetPriorities:207->transferAndWait:250->EmbeddedJournalIntegrationTestBase.waitForQuorumPropertySize:58 » Timeout

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**